packages/core/src/execute/BaseRoutine.ts

```diff
@@ -38,19 +38,34 @@ export default abstract class BaseRoutine<Ctx extends Context<BaseContextArgs>>
 
   async execute(context: Ctx): Promise<any[]> {
     const value = await this.serializeTasks();
-    const { other, priority } = this.orderByWorkspacePriorityGraph();
-
-    await this.serializeRoutines(value, priority);
+    const batches = this.orderByWorkspacePriorityGraph();
+    const allErrors: Error[] = [];
+    const allResults: any[] = [];
 
     const concurrency = context.args.concurrency || this.tool.config.execute.concurrency;
-    const response = await this.poolRoutines(value, concurrency ? { concurrency } : {}, other);
 
-    if (response.errors.length > 0) {
-      this.formatAndThrowErrors(response.errors);
+    // eslint-disable-next-line no-restricted-syntax
+    for (const batch of batches) {
+      // eslint-disable-next-line no-await-in-loop
```

Comment:
@milesj are these alright? I could rewrite it do some promise chaining stuff, but not sure how much it matters for cli tools.

Reply:
This is fine for now.